tools/run_tests/xds_k8s_test_driver/framework/test_app/client_app.py

```diff
@@ -230,6 +257,78 @@ def wait_for_server_channel_state(
         )
         return channel
 
+    def wait_for_xds_channel_active(
+        self,
+        *,
+        xds_server_uri: Optional[str] = None,
+        timeout: Optional[_timedelta] = None,
+        rpc_deadline: Optional[_timedelta] = None,
+    ) -> _ChannelzChannel:
+        # When polling for a state, prefer smaller wait times to avoid
+        # exhausting all allowed time on a single long RPC.
+        if rpc_deadline is None:
+            rpc_deadline = _timedelta(seconds=30)
+
+        retryer = retryers.exponential_retryer_with_timeout(
+            wait_min=_timedelta(seconds=10),
+            wait_max=_timedelta(seconds=25),
+            timeout=_timedelta(minutes=5) if timeout is None else timeout,
+        )
+
+        logger.info(
+            "[%s] Waiting to report an active channel to %s",
+            self.hostname,
+            xds_server_uri,
+        )
+        channel = retryer(
+            self.find_active_xds_channel,
+            xds_server_uri=xds_server_uri,
+            rpc_deadline=rpc_deadline,
+        )
+        logger.info(
+            "[%s] Channel to %s transitioned to active",
+            self.hostname,
+            xds_server_uri,
+        )
+        return channel
+
+    def find_active_xds_channel(
+        self,
+        *,
+        xds_server_uri: Optional[str] = None,
+        rpc_deadline: Optional[_timedelta] = None,
+    ) -> _ChannelzChannel:
+        rpc_params = {}
+        if rpc_deadline is not None:
+            rpc_params["deadline_sec"] = rpc_deadline.total_seconds()
+
+        for channel in self.get_server_channels(xds_server_uri, **rpc_params):
+            logger.info(
+                "[%s] xds channel: %s",
+                self.hostname,
+                _ChannelzServiceClient.channel_repr(channel),
+            )
+
+            try:
+                channel = self.check_channel_active(channel, **rpc_params)
+                logger.info(
```

Comment:
Instead of logging twice, let's add the `calls_started` and `calls_failed` to the `_ChannelzServiceClient.channel_repr(channel)`

Reply:
Make sense, I'll add `calls_started/failed` to `channel_repr` and remove this log.